### ISSUE TITLE
[Bug] lr is wrong when resume with an LinearLR

### ISSUE THREAD

**Issue by Cosmo1210** [open-mmlab/mmengine#1542]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmengine/issues) and [Discussions](https://github.com/open-mmlab/mmengine/discussions) but cannot get the expected help.
- [X] The bug has not been fixed in the latest version(https://github.com/open-mmlab/mmengine).

### Environment

NA

### Reproduces the problem - code sample

./tools/dist_train.sh configs/any_model_config.py 1 --resume $(any_checkpoint_iters_larger_than_LinearLR_end)

### Reproduces the problem - command or script

NA

### Reproduces the problem - error message

NA

### Additional information

in runner.resume(), runner build schedular first then load_state_dict():
https://github.com/open-mmlab/mmengine/blob/66fb81f7b392b2cd304fc1979d8af3cc71a011f5/mmengine/runner/runner.py#L2089-L2101

in the end of __init__() of _ParamScheduler(), it calls step()
https://github.com/open-mmlab/mmengine/blob/66fb81f7b392b2cd304fc1979d8af3cc71a011f5/mmengine/optim/scheduler/param_scheduler.py#L129

therefore the schedular calls step() before load_state_dict(), it will make LinearLR * start_factor to the lr which should not

**Comment by Cosmo1210** [comment]
just find my version is not the latest version, lastest one has fixed
